File: caesarCipher.c

# include <stdio.h>
# include <string.h>

void caesarCipher(char message[], int shift){
    for (int i =0; message[i] != '\0'; i++){
        char c = message[i];

        if(c>='a' && c<= 'z'){
            message[i] = (c-'a' - shift + 26)%26 +'a';
        }else if(c>='A' && c<='Z'){
            message[i] = (c-'A'-shift+26)%26+'A';
        }
    }
}

int main(){
    int N;
    char message[101];

    scanf("%d", &N);
    getchar();
    fgets(message, sizeof(message), stdin);

    int len = strlen(message);
    if(message[len-1] == '\n'){
        message[len-1] = '\0';
    }
    caesarCipher(message,N);
    printf("%s\n",message);
    return 0;
}

File: caesarCipher.swift

import Foundation

func caesarCipher(_ message: String, _ shift: Int)-> String {
    let shift = shift%26
    var encryptedMessage = ""

    for char in message{
        if let ascii = char.asciiValue{
            if char.isLowercase{
                let newAscii = (Int(ascii)-97-shift+26)%26+97
                encryptedMessage.append(Character(UnicodeScalar(newAscii)!))
            }else if char.isUppercase{
                let newAscii = (Int(ascii)-65-shift+26)%97+65
                encryptedMessage.append(Character(UnicodeScalar(newAscii)!))
            }else{
                encryptedMessage .append(char)
            }
        }else{
            encryptedMessage.append(char)
        }
    
}
    return encryptedMessage
}


if let N = Int(readLine() ?? "0"), let message = readLine(){
    print(caesarCipher(message, N))
}